### DISCUSSION TITLE
Add support for `Arrow` type.

### DISCUSSION THREAD

**Discussion by karolzlot** [pydantic/pydantic#3721] [Feature Request]
Currently it is possible to use `datetime` type for data and time. 

[Arrow](https://github.com/arrow-py/arrow) library makes dealing with dates and times a lot easier. It would be convenient to have support of it in Pydantic.  It would be great to have support for `Arrow` type.

`arbitrary_types_allowed = True` allows to use it, but it doesn't work well: https://stackoverflow.com/questions/68930298/how-to-use-arrow-type-in-fastapi-response-schema (solution presented here is the best what can be achieved as of now, but it is not ideal solution)

**Comment by rjurney** [comment]
It would be great from a performance perspective to allow direct integration with Arrow for loading data into objects.
